packages/kbn-esql-validation-autocomplete/src/shared/context.ts

```diff
@@ -151,6 +152,20 @@ function isBuiltinFunction(node: ESQLFunction) {
  * * "newCommand": the cursor is at the beginning of a new command (i.e. `command1 | command2 | <here>`)
  */
 export function getAstContext(queryString: string, ast: ESQLAst, offset: number) {
+  let inComment = false;
+
+  Walker.visitComments(ast, (node) => {
+    if (node.location.min <= offset && node.location.max > offset) {
+      inComment = true;
+    }
+  });
```

Comment:
@vadimkibana lmk if there's a better way...

Reply:
LGTM. The `Walker` will visit all comment nodes. We could make it more efficient: use binary search and zoom into specifically to the location of interest. But, I don't think we need that complication here if this is fast enough.